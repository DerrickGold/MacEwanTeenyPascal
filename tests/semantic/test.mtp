const longString123456789011 := 'This is a really super long string that may not look that great in the symbol table';

begin


end.