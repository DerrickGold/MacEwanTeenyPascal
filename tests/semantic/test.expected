
Semantically Corrected AST
[Head] (Block, Block Statement) 
Symbol Table:============================================================================:
Identifiers    |Stack Offset  |Type                 |Value                           
-----------------------------------------------------------------------------------------
longString1... |NONE          |Str, Con             |'This is a really super long... 
=======================================================================Depth 0   :Size 0   
    [Child] (Const Section) 
        [Child] (Const Declaration) 
            [Child] 1, TOK_ID, longString123456789011, (Literal) 
            [Child] 1, TOK_STR, 'This is a really super long string that may not look that great in the symbol table', (String) 
    [Child] (Statement List) 
        [Child] (Null Statement) 
